compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp

```diff
@@ -267,8 +268,17 @@ tryRaiseToExtractSlice(AffineMap inputIndexingMap, AffineMap outputIndexingMap,
   IntegerAttr one = rewriter.getI64IntegerAttr(1);
   unsigned currOutDim = 0;
   for (auto [idx, expr] : llvm::enumerate(inputIndexingMap.getResults())) {
+    // Assume that the constant access is a rank reducing access.
```

Comment:
This comment is now outdated and misleading. This now implicitly handles rank-reducing slicing. I think this should be mentioned instead.

Reply:
I think this is still what the code is doing though. As written, the linalg ops that are being raised are reading a slice of sizes `output_shape` from the `tensor.extract` target. Any constant access on the input is necessarily size 1, however if the output shape for that index is also 1 this is not required to be rank reducing.

We have to basically guess as to whether a constant access is rank reducing or not and then reconcile that as we walk the output map results. Currently the code requires all of the output map results to be accounted for with a non-rank reducing access (currOutDim only increments when it hits the associated AffineDimExpr) and thus all constant accesses must be rank reducing for the raising to succeed (because we require that the output indexing map is an identity). In other words, here we are first assuming that a constant access is rank reducing and then will fail to raise if that's not the case. The alternative case is part of what #14845 is trying to handle.

Reply:
I'm going to land this as is to unblock the linked issue. If I misinterpreted something here then it can be addressed in a follow up.